yelp_uri/linkify.py

```diff
@@ -0,0 +1,49 @@
+import re
+
+from yelp_uri.search import url_regex_common_tlds
+from yelp_uri.encoding import decode_uri
+from yelp_uri.encoding import encode_uri
+
+
+def split_linenumber(uri_match):
+    '''Split uri into two parts: the actual uri and the line number'''
+
+    path_query_fragment = uri_match.group('path_query_fragment')
+    line_number = ''
+    if path_query_fragment is not None:
+        _, line_number = re.match(r'^(.+?)((?::\d+)*)$', path_query_fragment).groups('')
```

Comment:
.groups('') is weird. I haven't seen that before.

Are you sure that `return match().groups()` isn't the same behavior?

I think `((?::\d+)*)` would be simpler as `(:[:0-9]+)`.

It's slightly better to make the regex object a global constant and reuse it.